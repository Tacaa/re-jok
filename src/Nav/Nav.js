// @flow
import * as React from 'react';
import {
  StyledNavBar,
  StyledNavBarContainer,
  StyledNavBarLeft,
  StyledNavBarRight,
  StyledNavBarCenter
} from './style';


type Props = {
  /** Children of Navbar ... This is element left content**/
  children?: any,
  /** This is element right content**/
  elementRight?: any,
  /** This is element center content... i just recommend use this to set logo**/
  elementCenter?: any,
  /** Set navbar transparent or not**/
  transparent: boolean,
  /** Set background color of navbar**/
  backgroundColor?: string,
}

const defaultProps = {
  transparent: false
}


class Nav extends React.Component<Props> {

  static defaultProps = defaultProps;

  render(){
    const {
      children,
      elementRight,
      elementCenter,
      ...rest
    } = this.props;
    return (
      <StyledNavBar {...rest}>
        <StyledNavBarContainer>
          <StyledNavBarLeft>
            {children}
          </StyledNavBarLeft>
          {elementCenter && <StyledNavBarCenter>{elementCenter}</StyledNavBarCenter>}
          {elementRight && <StyledNavBarRight>{elementRight}</StyledNavBarRight>}
        </StyledNavBarContainer>

      </StyledNavBar>
    )
  }
}

export default Nav
