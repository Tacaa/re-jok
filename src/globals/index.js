import styled from 'styled-components';

export const hexa = (hex, alpha) => {
  var r = parseInt(hex.slice(1, 3), 16),
    g = parseInt(hex.slice(3, 5), 16),
    b = parseInt(hex.slice(5, 7), 16);


  if (alpha >= 0) {
    return `rgba(${r}, ${g}, ${b}, ${alpha})`;
  } else {
    return `rgb(${r}, ${g}, ${b})`;
  }
};

export const shadeColor = (color, percent) => {
  if(color.charAt(0) !== "#") return color;
  var R = parseInt(color.slice(1, 3), 16),
    G = parseInt(color.slice(3, 5), 16),
    B = parseInt(color.slice(5, 7), 16);

    R = parseInt(R * (100 + percent) / 100, 10);
    G = parseInt(G * (100 + percent) / 100, 10);
    B = parseInt(B * (100 + percent) / 100, 10);

    R = (R<255)?R:255;
    G = (G<255)?G:255;
    B = (B<255)?B:255;

    var RR = ((R.toString(16).length===1)?"0"+R.toString(16):R.toString(16));
    var GG = ((G.toString(16).length===1)?"0"+G.toString(16):G.toString(16));
    var BB = ((B.toString(16).length===1)?"0"+B.toString(16):B.toString(16));

    return "#"+RR+GG+BB;
};



export const PrimaryText = styled.span `
  color: ${({theme}) => theme.text.primary};
  font-size: 1.5rem;
  font-weight: 500;
  line-height: 1.5rem;
`;

export const SecondaryText = styled.span `
  color:${({theme}) => theme.text.secondary};
  font-size: 1.2rem;
  font-weight: 200;
  line-height: 2.5rem;

`;
