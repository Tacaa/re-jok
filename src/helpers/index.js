export const generalId = () => {
  return '_' + Math.random().toString(36).substr(2, 9);
}

//Default Function
export const noop = () => {}
