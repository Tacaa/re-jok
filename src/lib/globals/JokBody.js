// @flow
import * as React from 'react';
import ReactDOM, { createPortal } from 'react-dom';
import Modal from '../Modal';
import Notification from '../Notification';
import Sidebar from '../Sidebar';



type Props = {
    /* Children shoube be Modal or Notification */
    children: React.Element < typeof Modal > | React.Element < typeof Notification > | React.Element < typeof Sidebar >,
}


class JokBody extends React.PureComponent<Props> {


    _popup: HTMLDivElement;

    constructor(props) {
       super(props);
       this._popup = document.createElement('div');
     }

    componentDidMount() {
        if (!document.body) throw new Error("Unexpectedly missing <body>.");
        const body: HTMLElement = document.body;
        this._popup = document.createElement('div');
        body.appendChild(this._popup);
    }

    componentWillUnmount() {
        if (!document.body) throw new Error("Unexpectedly missing <body>.");
        const body: HTMLElement = document.body;
        ReactDOM.unmountComponentAtNode(this._popup);
        body.removeChild(this._popup);
    }

    _render() {
        const { children, ...rest} = this.props;

        createPortal(
          <children.type {...children.props} {...rest} />
        , this._popup);
    }

    render() {
        const { children, ...rest} = this.props;

        return createPortal(
          <children.type {...children.props} {...rest} />
        , this._popup)
    }
}

export default JokBody
