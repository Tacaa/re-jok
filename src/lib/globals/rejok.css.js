//@flow
import { injectGlobal } from 'styled-components';
import { ClassFlex } from '../classes/flex';
import { ClassMargin } from '../classes/margin';
import { ClassPadding } from '../classes/padding';
import { ClassShadow } from '../classes/shadow';

export const globalConfig = () => injectGlobal`
@import url('https://fonts.googleapis.com/css?family=Roboto');

 *,
 *::after,
 *::before {
   margin: 0;
   padding: 0;
   box-sizing: inherit; }

 html {
   font-size: 62.5%; }

 body {
   font-family: "Roboto", "Helvetica", "Arial", sans-serif;
   font-weight: 400;
   box-sizing: border-box;
   line-height: 1.7;
   -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  }

  /* Support class*/


  /* class for flex */

  ${ClassFlex}

  /* class for margin */

  ${ClassMargin}

  /* class for padding */

  ${ClassPadding}

  /* class for box-shadow */

  ${ClassShadow}


`
