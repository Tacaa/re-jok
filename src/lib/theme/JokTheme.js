// @flow
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from './index';

type Props = {
  /**Theme of project**/
  theme: Object,
  /**Children can be anything**/
  children: any,
}

const defaultProps = {
  theme: {}
}

class JokTheme extends React.Component<Props>{
  static defaultProps= defaultProps;

  checkDefaultTheme = ()=>{
    const {theme} = this.props;
    if(Object.keys(theme).length === 0 && theme.constructor === Object) return true;
    return false;
  }

  render(){
    const {
      children,
      theme
    } = this.props;

    let useDefaultTheme = this.checkDefaultTheme();

    return (
      <ThemeProvider theme={useDefaultTheme? defaultTheme : theme}>
        {children}
      </ThemeProvider>
    )
  }
}

export default JokTheme
