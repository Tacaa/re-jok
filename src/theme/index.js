// @flow
// export our theme to the provider
// This is globally available in styled-components when interpolating a function like so:
// ${(props) => props.theme}
// Or using import { withTheme } from 'styled-components';


export const defaultTheme = {
  size: {
    small: {
      fontSize: '1.2rem',
      padding: '0.5rem 0.9rem',
      "lineHeight": "1",
      "borderRadius": ".2rem",
      width: "2.4rem",
      height: "2.4rem"
    },
    default: {
      fontSize: '1.4rem',
      padding: '0.9rem 1.8rem',
      lineHeight: "1.5",
      borderRadius: ".4rem",
      width: "3.2rem",
      height: "3.2rem"
    },
    medium: {
      fontSize: '1.8rem',
      padding: '1.4rem 2.7rem',
      lineHeight: "1.5",
      borderRadius: ".6rem",
      width: "4rem",
      height: "4rem"

    },
    large: {
      fontSize: '2rem',
      padding: '1.8rem 3.6rem',
      lineHeight: "2",
      borderRadius: ".6rem",
      width: "4.8rem",
      height: "4.8rem"
    }
  },
  grid:{
    xs: 768,
    sm: 768,
    md: 992,
    lg: 1200,
    xl: 1920
  },
  text: {
    primary: "rgba(0, 0, 0, 0.85)",
    secondary: "rgba(0, 0, 0, 0.48)",
    hint: "rgba(0, 0, 0, 0.38)"
  },
  background: {
    paper: "#fff",
    default: "#fafafa"
  },
  typography: {
    title: {
      fontSize: "2rem",
      fontWeight: "600",
      lineHeight: "2.8rem"

    },
    subtitle: {
      fontSize: "1.6rem",
      fontWeight: "400",
      lineHeight: "2.4rem"
    }
  },
  primary: {
    main: "#1c8ffb",

  },
  secondary: {
    main: "#1b1c1d",
  },
  info: {
    main: "#1890ff"
  },
  error: {
    main: "#f5222d"
  },
  success: {
    main: "#52c41a"
  },
  warning: {
    main: "#faad14"
  },
  zIndex: {
    base: 1,
    background: 0,
    hidden: -1,
    card: 2,
    loading: 3,
    avatar: 3,
    form: 3,
    search: 3,
    badge: 100,
    modal: 1500,
  }
};

const changeTheme = (defaultTheme, theme)=>{
  let cloneTheme = {...defaultTheme};


  Object.keys(theme).forEach(function (key) {
    if ( 'object' === typeof theme[key] && !Array.isArray(theme[key]) ) {
    	changeTheme(cloneTheme[key], theme[key])
    }
    else if ( 'string' === typeof theme[key] || 'number' === typeof theme[key] ) {
      cloneTheme[key] = theme[key]
    }
  })
  return cloneTheme;
}

export const setTheme = (theme: Object) => {

   const newTheme = changeTheme(defaultTheme, theme);

   return newTheme;

}
