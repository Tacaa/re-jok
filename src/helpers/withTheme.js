import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import {defaultTheme} from '../theme';
import { ThemeProvider } from 'styled-components';
export function renderWithTheme(component) {

  return renderer.create(
    <ThemeProvider theme={defaultTheme}>
      {component}
    </ThemeProvider>
  );
};

export const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
};

export const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  })
};
