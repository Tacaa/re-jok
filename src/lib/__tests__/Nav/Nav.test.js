import React from 'react';
import Nav from '../../Nav/Nav';
import { hexa } from  '../../globals';
import {defaultTheme} from '../../theme';

import {shallowWithTheme, mountWithTheme, renderWithTheme} from '../../helpers/withTheme';
import 'jest-enzyme';
import 'jest-styled-components';

describe('<Nav />', () => {
  it('should render Nav', () => {

    const tree = renderWithTheme(<Nav>
        Left Element
      </Nav>).toJSON();


    expect(tree).toMatchSnapshot();


  });

  it('should render Nav with primary color', () => {

   const wrapper = mountWithTheme(<Nav primary>Left Element</Nav>, defaultTheme);

   expect(wrapper.props().primary).toEqual(true);

   const tree = renderWithTheme(<Nav primary>
       Left Element
     </Nav>).toJSON();
     expect(tree).toMatchSnapshot();

     expect(tree).toHaveStyleRule('background-color', defaultTheme.primary.main);
     expect(tree).toHaveStyleRule('color', defaultTheme.text.default);



  });


  it('should render Nav with secondary color', () => {

   const wrapper = mountWithTheme(<Nav secondary>Left Element</Nav>, defaultTheme);

   expect(wrapper.props().secondary).toEqual(true);

   const tree = renderWithTheme(<Nav secondary>
       Left Element
     </Nav>).toJSON();
     expect(tree).toMatchSnapshot();

     expect(tree).toHaveStyleRule('background-color', defaultTheme.secondary.main);
     expect(tree).toHaveStyleRule('color', defaultTheme.text.default);



  });


  it('should render Nav with background color', () => {

   const wrapper = mountWithTheme(<Nav backgroundColor="#bababa">Left Element</Nav>, defaultTheme);

   expect(wrapper.props().backgroundColor).toEqual("#bababa");

   const tree = renderWithTheme(<Nav backgroundColor="#bababa">
       Left Element
     </Nav>).toJSON();
     expect(tree).toMatchSnapshot();

     expect(tree).toHaveStyleRule('background-color', "#bababa");
     expect(tree).toHaveStyleRule('color', defaultTheme.text.default);



  });

  it('should render Nav with transparent background', () => {

   const wrapper = mountWithTheme(<Nav transparent>Left Element</Nav>, defaultTheme);

   expect(wrapper.props().transparent).toEqual(true);

   const tree = renderWithTheme(<Nav transparent>
       Left Element
     </Nav>).toJSON();
     expect(tree).toMatchSnapshot();

     expect(tree).toHaveStyleRule('background', "transparent");
     expect(tree).toHaveStyleRule('color', 'rgba(0,0,0,.64)');



  });



});
