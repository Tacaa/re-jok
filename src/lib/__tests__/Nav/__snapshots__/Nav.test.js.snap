// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render Nav 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding-right: 1.5rem;
  padding-left: 1.5rem;
  margin-right: auto;
  margin-left: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  min-height: 6.4rem;
  font-size: 1.4rem;
  padding: 1rem 0;
  z-index: 400;
  background-color: #fff;
  color: rgba(0,0,0,.64);
}

.c0 .sc-bZQynM {
  display: none;
}

@media (max-width:1000px) {
  .c0 .sc-ifAKCX {
    display: none;
  }

  .c0 .sc-EHOje {
    display: none;
  }

  .c0 .sc-bZQynM {
    display: block;
  }
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Left Element
    </div>
  </div>
</nav>
`;

exports[`<Nav /> should render Nav with background color 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding-right: 1.5rem;
  padding-left: 1.5rem;
  margin-right: auto;
  margin-left: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  min-height: 6.4rem;
  font-size: 1.4rem;
  padding: 1rem 0;
  z-index: 400;
  background-color: #bababa;
  color: #fff;
}

.c0 .sc-bZQynM {
  display: none;
}

@media (max-width:1000px) {
  .c0 .sc-ifAKCX {
    display: none;
  }

  .c0 .sc-EHOje {
    display: none;
  }

  .c0 .sc-bZQynM {
    display: block;
  }
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Left Element
    </div>
  </div>
</nav>
`;

exports[`<Nav /> should render Nav with primary color 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding-right: 1.5rem;
  padding-left: 1.5rem;
  margin-right: auto;
  margin-left: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  min-height: 6.4rem;
  font-size: 1.4rem;
  padding: 1rem 0;
  z-index: 400;
  background-color: #1c8ffb;
  color: #fff;
}

.c0 .sc-bZQynM {
  display: none;
}

@media (max-width:1000px) {
  .c0 .sc-ifAKCX {
    display: none;
  }

  .c0 .sc-EHOje {
    display: none;
  }

  .c0 .sc-bZQynM {
    display: block;
  }
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Left Element
    </div>
  </div>
</nav>
`;

exports[`<Nav /> should render Nav with secondary color 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding-right: 1.5rem;
  padding-left: 1.5rem;
  margin-right: auto;
  margin-left: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  min-height: 6.4rem;
  font-size: 1.4rem;
  padding: 1rem 0;
  z-index: 400;
  background-color: #1b1c1d;
  color: #fff;
}

.c0 .sc-bZQynM {
  display: none;
}

@media (max-width:1000px) {
  .c0 .sc-ifAKCX {
    display: none;
  }

  .c0 .sc-EHOje {
    display: none;
  }

  .c0 .sc-bZQynM {
    display: block;
  }
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Left Element
    </div>
  </div>
</nav>
`;

exports[`<Nav /> should render Nav with transparent background 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding-right: 1.5rem;
  padding-left: 1.5rem;
  margin-right: auto;
  margin-left: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  min-height: 6.4rem;
  font-size: 1.4rem;
  padding: 1rem 0;
  z-index: 400;
  background: transparent;
  color: rgba(0,0,0,.64);
}

.c0 .sc-bZQynM {
  display: none;
}

@media (max-width:1000px) {
  .c0 .sc-ifAKCX {
    display: none;
  }

  .c0 .sc-EHOje {
    display: none;
  }

  .c0 .sc-bZQynM {
    display: block;
  }
}

<nav
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Left Element
    </div>
  </div>
</nav>
`;
